lib/rules/operator-linebreak.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @fileoverview Operator linebreak - enforces operator linebreak style of two types: after and before
+ * @author Benoît Zugmeyer
+ * @copyright 2015 Benoît Zugmeyer. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+
+    var style = context.options[0] || "after";
+
+    //--------------------------------------------------------------------------
+    // Helpers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Checks whether two tokens are on the same line.
+     * @param {ASTNode} left The leftmost token.
+     * @param {ASTNode} right The rightmost token.
+     * @returns {boolean} True if the tokens are on the same line, false if not.
+     * @private
+     */
+    function isSameLine(left, right) {
+        return left.loc.end.line === right.loc.start.line;
+    }
+
+    /**
+     * Checks the operator placement
+     * @param {ASTNode} node The binary operator node to check
+     * @private
+     * @returns {void}
+     */
+    function validateBinaryExpression(node) {
+        var leftToken = context.getLastToken(node.left || node.id);
+        var operatorToken = context.getTokenAfter(leftToken);
+        var rightToken = context.getTokenAfter(operatorToken);
+        var operator = operatorToken.value;
```

Comment:
```js
(a)
+
(b)
```

Unfortunately, this is another place we run into the case where parentheses around expressions mess up tokens. The snippet above lints without error because `lastToken`, `operatorToken`, and `rightToken` are `a`, `)`, and `+`, respectively. There needs to be a check to make sure it actually gets the correct token for the operator. Once that's done, the tokens on its left and right side are trivial to retrieve.